Arf; linting failure

```
daemon/containerd/platform_matchers.go:28:6: type `matchComparerProvider` is unused (unused)
type matchComparerProvider func(ocispec.Platform) platforms.MatchComparer
     ^
daemon/containerd/platform_matchers.go:30:24: func `(*ImageService).matchRequestedOrDefault` is unused (unused)
func (i *ImageService) matchRequestedOrDefault(
                       ^

--- a/daemon/containerd/platform_matchers.go
+++ b/daemon/containerd/platform_matchers.go
@@ -24,3 +24,11 @@ func (c allPlatformsWithPreferenceMatcher) Match(_ ocispec.Platform) bool {
 func (c allPlatformsWithPreferenceMatcher) Less(p1, p2 ocispec.Platform) bool {
 	return c.preferred.Less(p1, p2)
 }
+
+func (i *ImageService) hostPlatformMatcher() platforms.MatchComparer {
+	// Allow to override the host platform for testing purposes.
+	if i.defaultPlatformOverride != nil {
+		return i.defaultPlatformOverride
+	}
+	return platforms.Default()
+}

